packages/medusa/src/repositories/product-tag.ts

```diff
@@ -1,5 +1,56 @@
-import { EntityRepository, Repository } from "typeorm"
+import { EntityRepository, In, Repository } from "typeorm"
 import { ProductTag } from "../models/product-tag"
 
+type UpsertTagsInput = (Partial<ProductTag> & {
+  value: string
+})[]
+
 @EntityRepository(ProductTag)
-export class ProductTagRepository extends Repository<ProductTag> {}
+export class ProductTagRepository extends Repository<ProductTag> {
+  public async listTagsByUsage(count = 10): Promise<ProductTag[]> {
+    const tags = await this.query(
+      `
+        SELECT id, COUNT(pts.product_tag_id) as usage_count, pt.value
+        FROM product_tag pt
+        LEFT JOIN product_tags pts ON pt.id = pts.product_tag_id
+        GROUP BY id
+        ORDER BY usage_count DESC
+        LIMIT $1
+      `,
+      [count]
+    )
+
+    return tags
```

Comment:
suggestion: The tags assignment is not necessary here, we can directly return await ...